src/main/java/com/algolia/search/InsightsConfig.java

```diff
@@ -6,51 +6,66 @@
-   * @param apiKey The API Key
-   */
-  public InsightsConfig(String applicationID, String apiKey) {
-    this(applicationID, apiKey, "us");
-  }
+  public static class Builder extends ConfigBase.Builder<Builder> {
+
+    /**
+     * Creates an {@link InsightsConfig} with the default region ("us").
+     *
+     * @param applicationID The ApplicationID
+     * @param apiKey The API Key
+     */
+    public Builder(@Nonnull String applicationID, @Nonnull String apiKey) {
+      this(applicationID, apiKey, "us");
+    }
+
+    /**
+     * Creates an {@link InsightsConfig} with a custom region.
+     *
+     * @param applicationID The ApplicationID
+     * @param apiKey The API Key
+     */
+    public Builder(@Nonnull String applicationID, @Nonnull String apiKey, @Nonnull String region) {
+      super(applicationID, apiKey);
+
+      List<StatefulHost> hosts =
+          Collections.singletonList(
+              new StatefulHost(
+                  "insights." + region + ".algolia.io", EnumSet.of(CallType.READ, CallType.WRITE)));
+
+      this.setDefaultHosts(hosts);
+    }
+
+    /**
+     * Creates an {@link InsightsConfig} for the search client with custom {@link StatefulHost}.
+     * Warning: Defaults hosts are not set when setting custom {@link StatefulHost}.
+     *
+     * @param applicationID The ApplicationID
+     * @param apiKey The API Key
+     * @param customHosts List of custom hosts
+     */
+    public Builder(
+        @Nonnull String applicationID,
+        @Nonnull String apiKey,
+        @Nonnull List<StatefulHost> customHosts) {
+      super(applicationID, apiKey);
+
+      this.setCustomHosts(customHosts);
+    }
+
+    @Override
+    public Builder getThis() {
+      return this;
+    }
 
-  /**
-   * Creates an Insight client configuration with a custom region.
-   *
-   * @param applicationID The ApplicationID
-   * @param apiKey The API Key
-   * @param region Region of the server. For example "us"
-   */
-  public InsightsConfig(String applicationID, String apiKey, String region) {
-    super(applicationID, apiKey);
-
-    List<StatefulHost> hosts =
-        Collections.singletonList(
-            new StatefulHost(
-                "insights." + region + ".algolia.io", EnumSet.of(CallType.READ, CallType.WRITE)));
-
-    this.setDefaultHosts(hosts);
+    public InsightsConfig build() {
+      return new InsightsConfig(this);
+    }
   }
 
-  /**
-   * Creates an {@link InsightsConfig} for the search client with custom {@link StatefulHost}.
-   * Warning: Defaults hosts are not set when setting custom {@link StatefulHost}.
-   *
-   * @param applicationID The ApplicationID
-   * @param apiKey The API Key
-   * @param customHosts List of custom hosts
-   */
-  public InsightsConfig(
-      @Nonnull String applicationID,
-      @Nonnull String apiKey,
-      @Nonnull List<StatefulHost> customHosts) {
-    super(applicationID, apiKey);
-
-    this.setCustomHosts(customHosts);
+  private InsightsConfig(Builder builder) {
```

Comment:
Same as above: I suggest to keep the a private InsightConfig constructor with all parameters and invoke that in "build()"